tests/test_snmpcommunity.rb

```diff
@@ -34,45 +34,87 @@ class TestSnmpCommunity < CiscoTestCase
 
   def setup
     super
-    @default_show_command = 'show run snmp all | no-more'
+    if platform != :ios_xr
+      @default_show_command = 'show run snmp all | no-more'
+    else
+      @default_show_command = 'show running-config snmp'
+    end
   end
 
   def test_snmpcommunity_collection_empty
     # This test requires all the snmp communities removed from device
-    s = @device.cmd('show run snmp all | no-more')
-    cmd_prefix = 'snmp-server community'
-    # puts "s : #{s}"
-    pattern = /#{cmd_prefix}\s\S+\sgroup\s\S+/
-    until (md = pattern.match(s)).nil?
-      # puts "md : #{md}"
-      config("no #{md}")
-      s = md.post_match
+    if platform != :ios_xr
+      s = @device.cmd('show run snmp all | no-more')
+      cmd_prefix = 'snmp-server community'
+      # puts "s : #{s}"
+      pattern = /#{cmd_prefix}\s\S+\sgroup\s\S+/
+      until (md = pattern.match(s)).nil?
+        # puts "md : #{md}"
+        config("no #{md}")
+        s = md.post_match
+      end
+    else
+      config('no snmp-server community com1',
```

Comment:
This cleanup section won't be very reliable if other communities are setup on the test device.  I would prefer you use logic similar to nexus above where you use the 'show running' command go get all existing communities and remove them or use the class method to gather all communities and destroy them in the setup.